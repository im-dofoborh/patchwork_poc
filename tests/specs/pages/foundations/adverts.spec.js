import { mount, createLocalVue } from '@vue/test-utils';
import AdvertsPage from '~/pages/foundations/adverts';

describe('Adverts page', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('adverts-page', AdvertsPage);
    wrapper = mount(AdvertsPage, {
      localVue,
    });
  })

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
