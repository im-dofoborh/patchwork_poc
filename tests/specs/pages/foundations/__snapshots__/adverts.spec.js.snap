// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adverts page renders 1`] = `
<section class="sg-adverts-page">
  <div class="sg-content container">
    <div class="sg-content__header">
      <h1 class="heading-1">Adverts</h1>
    </div>
    <div class="sg-content__section">
      <h2 class="heading-2">Standard advert sizes</h2>
      <p>
        These are the standard fixed-size ad units available on Fabric. The markup
        below is a sample of the typical usage of one of these ad units, in this case
        the second MPU unit on a home page. Depending on the location, this markup
        could include different position names for ad targeting or different classes
        to allow for functionality like sticky advert behaviour to be applied with JavaScript.
      </p>
      <div class="sg-adverts-page__grid">
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 250px; width: 970px;">
            <h6 class="heading-6">Billboard</h6>
            <h6 class="heading-6">970x250</h6>
            <!---->
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 90px; width: 728px;">
            <h6 class="heading-6">Leaderboard</h6>
            <h6 class="heading-6">728x90</h6>
            <!---->
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 600px; width: 300px;">
            <h6 class="heading-6">Double MPU</h6>
            <h6 class="heading-6">300x600</h6>
            <!---->
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 250px; width: 300px;">
            <h6 class="heading-6">Single MPU</h6>
            <h6 class="heading-6">300x250</h6>
            <p class="body-copy-small">
              Also referred to as 300x251 and 300x252 to enable specific campaign targeting of individual MPUs when two are on the page
            </p>
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 125px; width: 300px;">
            <h6 class="heading-6">Mobile Mega Banner</h6>
            <h6 class="heading-6">300x125</h6>
            <!---->
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 50px; width: 300px;">
            <h6 class="heading-6">Mobile Banner</h6>
            <h6 class="heading-6">300x50</h6>
            <!---->
          </div>
        </div>
        <div class="sg-adverts-page__grid-item">
          <div class="sg-adverts-page__mock-advert" style="height: 50px; width: 320px;">
            <h6 class="heading-6">Mobile Banner</h6>
            <h6 class="heading-6">320x50</h6>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div class="sg-content__section editor-content">
      <h2>Additional advert types</h2>
      <p>As well as the standard advert types, we also serve the following ad units:</p>
      <h3>Out of page unit (1x1)</h3>
      <p>
        On devices which are wide enough, we use an ad unit of the size "1x1" to display
        an advert which wraps around the top and sides of the page grid. This is known
        as a takeover advert. When this happens, the advert should append the
        <code>.ad-skin-active</code> class to the <code>body</code> element to allow us
        to control the page styling and ensure any full-width elements, such as
        the site header, fit neatly to the grid.
      </p>
      <p>
        A series of utility classes, including <code>.fluid-container</code> and
        <code>.body-background</code> are also available and can be used on markup to ensure
        the page responds correctly when a takeover advert is served. The
        <a href="https://immediateco.atlassian.net/wiki/spaces/WCPP/pages/5181216/Advert+integrations">Advert Integrations</a>
        page on Confluence documents these classes in more detail.
      </p>
      <h3>Sharethrough unit (5x5)</h3>
      <p>
        A Sharethrough ad unit is a native ad unit which can be used within the content
        of a page, such as within an archive page. This unit requires a Sharethrough key
        which is unique based on the site and type of location (archive, article etc).
        A Sharethrough key is usually configured within the "Site Config" screen by
        a site admin. Including the Sharethrough Twig partial and passing the correct key
        will ensure your unit displays correctly.
      </p>
    </div>
    <div class="sg-content__section editor-content">
      <h2>Sticky adverts</h2>
      <p>
        On some pages we use JavaScript to enable sticky advert behaviour to ensure an advert
        remains on screen between specified scroll positions. Full documentation
        on this behaviour can be found on the
        <a href="https://immediateco.atlassian.net/wiki/spaces/WCPP/pages/5181005/Sticky+Advert+Behaviour">Sticky Advert Behaviour</a>
        Confluence page.
      </p>
    </div>
  </div>
</section>
`;
